GETTING_STARTED.md

```diff
@@ -0,0 +1,71 @@
+# Opening your first Pull Request
+
+If you'd like to get familiar with xDSL, and are not sure how to get started, this document will walk you through opening your first pull request in the repository.
+When a new member joins the team, we ask them to perform the following series of steps on the first day, and it takes complete novices at most a couple of hours from starting this task to opening a PR.
+One of the ways to make the time as short as possible is being in the same room as someone familiar with the project.
+The next best thing to asking questions in person is reaching out on [our Zulip chatroom](https://xdsl.zulipchat.com).
+Please don't hesitate to post to the Beginners channel with any questions about the framework.
+
+## Pre-Requisites
+
+Before starting the coding part of this tutorial please make sure that xDSL is correctly installed.
+The instructions on setting it up locally are in the [README](README.md).
+In particular, please follow the steps in **Developer Installation**, and **Formatting and Typechecking**.
+Please run `make tests` in the terminal.
+If they pass, you're ready to start!
+(If they don't, and you're not sure why, please reach out on the Zulip.)
+
+## The Mission
+
+The aim of this tutorial is to add an optimisation to our RISC-V representation.
+Here are some examples of existing optimisations at a high level:
+
+``` txt
+x + 0 -> x
+2 + 2 -> 4
+x * 1 -> x
+```
+
+In xDSL, this class of optimisation is called "canonicalization".
+This kind of optimisation is common in compilers, and can broadly be described as a transformation that reduces the complexity of the IR, making it easier for further analysis to improve the program.
+For further information about canonicalization, please take a look at [MLIR's documentation](https://mlir.llvm.org/docs/Canonicalization/).
+
+## A Short Guide to xDSL
+
+We have a short series of notebooks describing the APIs necessary to build representations of code, and transformations on it.
+The notebooks in [docs/Toy](docs/Toy/) are the best place to start, we recommend first looking at them, and then at the implementation of the Toy compiler itself: [Dialect](docs/Toy/toy/dialects/toy.py), [Canonicalization](docs/Toy/toy/rewrites/optimise_toy.py).
+
+## Adding Your Own Canonicalization Pattern
+
+Here are some examples of first PRs from existing contributors:
+[#1566](https://github.com/xdslproject/xdsl/pull/1566),
+[#1567](https://github.com/xdslproject/xdsl/pull/1567).
+
+The first step is finding a missing optimisation pattern.
+You're welcome to come up with your own, or do one of the following:
+
+- `x * 2^k -> x << k`
+- `x & 0 -> 0`
+- `x | 0 -> x`
+- `x ^ 0 -> x`
```

Comment:
Does this risk to not age well? How do we come up with new work once this is completed?

Reply:
I mostly agree with this, but the way I see it is to have something like a "Hello, world!" in xDSL.
There is no novelty in "Hello, world!"  apart from serving as a "kicking the tires" activity.

This might be good for now, but we might need to revise to have a very simple "toy"-like tutorial (e.g., a `foo` dialect with 2 arith-like ops and a simple transform with one of them).

Reply:
Maybe. It's also interesting that the "hello world" example is actually useful, I'm sure it makes things more exciting. But generating an infinite stream of simple, useful work is difficult.

Reply:
I'd say that it would be worth the effort. The good news is that we're already adding x86, and I'll be surprised if we never add wasm or ARM dialects, so we're unlikely to run out of trivial optimisations to add.